x/airdrop/keeper/keeper_test.go

```diff
@@ -164,6 +166,84 @@ func (s *KeeperTestSuite) TestFeeDrip() {
 	s.Require().Equal(feeCollectorBalance, sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(4040)))
 }
 
+func (s *KeeperTestSuite) TestAddNewFundWithDiffDenom() {
+	amount := sdk.NewInt64Coin("denom", 4000)
+	s.Require().NoError(s.app.BankKeeper.SetBalance(s.ctx, addr1, amount)) // ensure the account is funded
+
+	addrBalance := s.app.BankKeeper.GetBalance(s.ctx, addr1, "denom")
+	moduleBalance := s.app.BankKeeper.GetBalance(s.ctx, moduleAddress, "denom")
+
+	// sanity check
+	s.Require().Equal(addrBalance, sdk.NewCoin("denom", sdk.NewInt(4000)))
+	s.Require().Equal(moduleBalance, sdk.NewCoin("denom", sdk.NewInt(0)))
+
+	fund := types.Fund{
+		Amount:     &amount,
+		DripAmount: sdk.NewInt(40),
+	}
+	s.Require().NoError(s.app.AirdropKeeper.AddFund(s.ctx, addr1, fund))
+
+	addrBalance = s.app.BankKeeper.GetBalance(s.ctx, addr1, "denom")
+	moduleBalance = s.app.BankKeeper.GetBalance(s.ctx, moduleAddress, "denom")
+
+	s.Require().Equal(addrBalance, sdk.NewCoin("denom", sdk.NewInt(0)))
+	s.Require().Equal(moduleBalance, sdk.NewCoin("denom", sdk.NewInt(4000)))
+}
+
+func (s *KeeperTestSuite) TestMultiDenomFeeDrip() {
+	amountStake := sdk.NewInt64Coin(sdk.DefaultBondDenom, 4000)
+	amountDenom := sdk.NewInt64Coin("denom", 4000)
+	fund1 := types.Fund{
+		Amount:     &amountStake,
+		DripAmount: sdk.NewInt(40)}
+
+	fund2 := types.Fund{
+		Amount:     &amountStake,
+		DripAmount: sdk.NewInt(4000), // will only last one block
```

Comment:
It would be a slightly better test if we didn't use the same 4000 40 combination but mixed it up a little